Azure/azure-cli
# [Storage] `az storage blob/container`: Support `--account-name` for non-standard account URL

**Related command**
`az storage container`

**Description**

When use non-standard account URL, account name can't be parsed from URL, thus `ValueError: Unable to determine account name for shared key credential` is raised. In this PR, it will use credential dict if `account_name` is provided.

Close https://github.com/Azure/azure-cli/issues/23702. 

**Testing Guide**
`az storage account create --name <name> --resource-group <rg> --edge-zone microsoftrrdclab1 -l eastus2euap`
`az storage container create -n test --blob-endpoint <blob-endpoint> --account-name <name> --account-key <account-key>`

---

This checklist is used to make sure that common guidelines for a pull request are followed.
